### PR TITLE
Fix 35_SignedDistanceFieldText sample

### PR DESCRIPTION
**Pull request by elix22**
Fix 35_SignedDistanceFieldText sample .
On some platforms such as Emscripten,macOS and iOS the text is not shown , this P.R. fixes this issue .
The built-in derivative functions dFdx, dFdy, and fwidth are optional and  must be enabled by the extension GL_OES_standard_derivatives .
This extention is not enabled by default on some platforms , for more information see https://www.khronos.org/registry/OpenGL/extensions/OES/OES_standard_derivatives.txt "

### CODE REVIEW

**Comment by elix22** [comment]
When is this fix going to be merged ?

**Comment by elix22** [comment]
Hi , when is this fix going to be merged into master ?

**Comment by weitjong** [comment]
Thanks for your patience. Probably after I have merged the GLES 3 work from orefkov or you can try to get help from other maintainers to review and merge.

**Comment by elix22** [comment]
It can wait , just wanted to know about the status

**Comment by elix22** [comment]
Is there any update on this ?

File: Source/Urho3D/Graphics/Graphics.h [hunk 1]
```diff
@@ -527,6 +527,9 @@ class URHO3D_API Graphics : public Object
     void SetVBO(unsigned object);
     /// Bind a UBO, avoiding redundant operation. Used only on OpenGL.
     void SetUBO(unsigned object);
+    
+    /// Return whether  the extention GL_OES_standard_derivatives is supported
```
Comment (weitjong) [inline Source/Urho3D/Graphics/Graphics.h]: Doxygen comment should be terminated with a period.

File: Source/Urho3D/Graphics/Graphics.h [hunk 2]
```diff
@@ -795,7 +798,8 @@ class URHO3D_API Graphics : public Object
     String orientations_;
     /// Graphics API name.
     String apiName_;
-
+    /// indicates if the extention GL_OES_standard_derivatives is supported
```
Comment (weitjong) [inline Source/Urho3D/Graphics/Graphics.h]: It should be started with initial cap.

File: Source/Urho3D/Graphics/OpenGL/OGLShaderVariation.cpp [hunk 3]
```diff
@@ -103,6 +103,19 @@ bool ShaderVariation::Create()
...
     String shaderCode;
+    /*
+        Relevant only to Fragment processing functions  , only in case of OpenGL ES 2.0  .
+        The built-in derivative functions dFdx, dFdy, and fwidth are optional, and
+        must be enabled by #extension GL_OES_standard_derivatives : enable
+        See : https://www.khronos.org/registry/OpenGL/extensions/OES/OES_standard_derivatives.txt
+        On some platforms it's already enabled by default but on some it's not .
+    */
+#if defined(GL_ES_VERSION_2_0)
+    if(type_ == PS && graphics_->glOESStandardDerivativesSupport() == true)
```
Comment (weitjong) [inline Source/Urho3D/Graphics/OpenGL/OGLShaderVariation.cpp]: spacing convention

File: Source/Urho3D/Graphics/Graphics.h (hunk 1 above)
Comment (1vanK) [reply Source/Urho3D/Graphics/Graphics.h]: `whether  the` (double space)

File: Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp [hunk 4]
```diff
@@ -2843,6 +2843,9 @@ void Graphics::CheckFeatureSupport()
 
     // Consider OpenGL shadows always hardware sampled, if supported at all
     hardwareShadowSupport_ = shadowMapFormat_ != 0;
+    
+    // check whether  the extention GL_OES_standard_derivatives is supported
```
Comment (1vanK) [inline Source/Urho3D/Graphics/OpenGL/OGLGraphics.cpp]: double space `whether  the`

**Comment by 1vanK** [comment]
https://github.com/urho3d/Urho3D/commit/45f3f12072a39c73e80bdb95e8f537b4a9bf0ee2

**Comment by 1vanK** [comment]
This PR enable extension for all shaders and it can affect performance, so I just added `#extension GL_OES_standard_derivatives : enable` directly to text shader.

**Pull request closed without merging**